File: flowcore/src/main/java/com/raizlabs/android/dbflow/annotation/QueryModel.java
package com.raizlabs.android.dbflow.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Description: Marks a Model class as NOT a {@link Table}, but generates code for retrieving data from a
 * generic query
 */
@Retention(RetentionPolicy.SOURCE)
@Target(ElementType.TYPE)
public @interface QueryModel {

    /**
     * @return Specify the name of the database to use.
     */
    String databaseName();

    /**
     * @return If true, all accessible, non-static, and non-final fields are treated as columns.
     * @see Table#allFields()
     */
    boolean allFields() default false;
}

File: core/src/main/kotlin/com/dbflow5/annotation/QueryModel.kt
package com.dbflow5.annotation

import kotlin.reflect.KClass

/**
 * Description: Marks a Model class as NOT a [Table], but generates code for retrieving data from a
 * generic query
 */
@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
annotation class QueryModel(
        /**
         * @return Specify the class of the database to use.
         */
        val database: KClass<*>,
        /**
         * @return If true, all accessible, non-static, and non-final fields are treated as valid fields.
         * @see Table.allFields
         */
        val allFields: Boolean = true,

        /**
         * @return If true, we throw away checks for column indexing and simply assume that the cursor returns
         * all our columns in order. This may provide a slight performance boost.
         */
        val orderedCursorLookUp: Boolean = false,
        /**
         * @return When true, we reassign the corresponding Model's fields to default values when loading
         * from cursor. If false, we assign values only if present in Cursor.
         */
        val assignDefaultValuesFromCursor: Boolean = true)
